### DISCUSSION TITLE
Dependency resolution fails with Tycho >= 2.3.0

### DISCUSSION THREAD

**Discussion by AhmetBilgin** [eclipse-tycho/tycho#361] [Q&A]
Hi,

**Scenario**
recently I tried to update our targetdefinition to Eclipse 4.21 (2021-09) and therefore added the new url (http://download.eclipse.org/releases/2021-09 via p2 Proxy in nexus) and updated the included feature versions. The targetdefinition worked in the Eclipse-PDE IDE build.
But I had problems with the headless build. I don't know if the problems are tycho related or nexus related. So I'm just going to share my findings:

**Reproducibility**
I can reproduce this with the [tycho rcp demo project](https://github.com/eclipse/tycho/tree/master/demo/itp04-rcp) when using a nexus p2 proxy for the [eclipse-2021-9 release](http://download.eclipse.org/releases/2021-09) and tycho version >= 2.3.0.

**What does not work:**
 - Tycho **>=2.3.0** and URL: (Our internal **Nexus P2 Proxy**) http://nexus-server/nexus/content/repositories/Eclipse-2021-09-R

**What works:**
I also tried Tycho version 2.2.0 as I have read in the release notes that version 2.3.0 has a stricter dependency resolution.
 - Tycho **2.2.0** and URL: (our internal **Nexus P2 Proxy**) http://nexus-server/nexus/content/repositories/Eclipse-2021-09-R
   - Nexus 2.12
 - Tycho **>= 2.2.0** and URL: https://download.eclipse.org/releases/2021-09
 - Tycho **>= 2.2.0** and mirroring with b3aggregator and serving via nexus

**Exception I'm getting when using tycho >= 2.3.0 and the nexus proxy of Eclipse-2021-09:**
<details>
  <summary>Click to expand stacktrace!</summary>

> $ mvn clean verify
> [INFO] Scanning for projects...
> [INFO] Computing target platform for MavenProject: example.group:eclipse-repository:1.0.0-SNAPSHOT @ /data/home/xyz/projects/tycho/demo/itp04-rcp/eclipse-repository/pom.xml
> [INFO] Resolving dependencies of MavenProject: example.group:eclipse-repository:1.0.0-SNAPSHOT @ /data/home/xyz/projects/tycho/demo/itp04-rcp/eclipse-repository/pom.xml
> [INFO] Resolving class path of MavenProject: example.group:eclipse-repository:1.0.0-SNAPSHOT @ /data/home/xyz/projects/tycho/demo/itp04-rcp/eclipse-repository/pom.xml
> [INFO] Computing target platform for MavenProject: example.group:example-bundle:0.1.0-SNAPSHOT @ /data/home/xyz/projects/tycho/demo/itp04-rcp/example-bundle/pom.xml
> [INFO] Resolving dependencies of MavenProject: example.group:example-bundle:0.1.0-SNAPSHOT @ /data/home/xyz/projects/tycho/demo/itp04-rcp/example-bundle/pom.xml
> [INFO] Resolving class path of MavenProject: example.group:example-bundle:0.1.0-SNAPSHOT @ /data/home/xyz/projects/tycho/demo/itp04-rcp/example-bundle/pom.xml
> [ERROR] Internal error: java.lang.RuntimeException: org.osgi.framework.BundleException: Bundle example-bundle cannot be resolved:example-bundle [68]
> [ERROR]   Unresolved requirement: Require-Bundle: org.eclipse.ui; bundle-version="3.6.0"
> [ERROR]     -> Bundle-SymbolicName: org.eclipse.ui; bundle-version="3.119.100.v20210830-0942"; singleton:="true"
> [ERROR]        org.eclipse.ui [5]
> [ERROR]          Unresolved requirement: Require-Bundle: org.eclipse.ui.workbench; bundle-version="[3.120.0,4.0.0)"; visibility:="reexport"
> [ERROR]            -> Bundle-SymbolicName: org.eclipse.ui.workbench; bundle-version="3.123.0.v20210817-0704"; singleton:="true"
> [ERROR]               org.eclipse.ui.workbench [64]
> [ERROR]                 Unresolved requirement: Import-Package: org.eclipse.e4.ui.internal.workbench.addons
> [ERROR]                   -> Export-Package: org.eclipse.e4.ui.internal.workbench.addons; bundle-symbolic-name="org.eclipse.e4.ui.workbench"; bundle-version="1.13.0.v20210521-0741"; version="0.0.0"; x-internal:="true"
> [ERROR]                      org.eclipse.e4.ui.workbench [50]
> [ERROR]                        Unresolved requirement: Require-Bundle: org.eclipse.e4.ui.model.workbench; bundle-version="1.2.0"
> [ERROR]                          -> Bundle-SymbolicName: org.eclipse.e4.ui.model.workbench; bundle-version="2.2.0.v20210727-1533"; singleton:="true"
> [ERROR]                             org.eclipse.e4.ui.model.workbench [48]
> [ERROR]                               Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
> [ERROR]                                 -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
> [ERROR]                                    org.eclipse.e4.core.services [32]
> [ERROR]                                      Unresolved requirement: Require-Capability: osgi.extender; filter:="(&(osgi.extender=osgi.component)(version>=1.2)(!(version>=2.0)))"
> [ERROR]                        Unresolved requirement: Require-Bundle: org.eclipse.e4.ui.services; bundle-version="0.9.0"
> [ERROR]                          -> Bundle-SymbolicName: org.eclipse.e4.ui.services; bundle-version="1.5.0.v20210115-1333"; singleton:="true"
> [ERROR]                             org.eclipse.e4.ui.services [41]
> [ERROR]                               Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
> [ERROR]                                 -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
> [ERROR]                        Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
> [ERROR]                          -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
> [ERROR]                 Unresolved requirement: Import-Package: org.eclipse.e4.core.commands
> [ERROR]                   -> Export-Package: org.eclipse.e4.core.commands; bundle-symbolic-name="org.eclipse.e4.core.commands"; bundle-version="1.0.0.v20210507-1901"; version="0.0.0"
> [ERROR]                      org.eclipse.e4.core.commands [28]
> [ERROR]                        Unresolved requirement: Import-Package: org.eclipse.e4.core.services.log
> [ERROR]                          -> Export-Package: org.eclipse.e4.core.services.log; bundle-symbolic-name="org.eclipse.e4.core.services"; bundle-version="2.3.0.v20210723-1034"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.bindings,  org.eclipse.e4.ui.services,  org.eclipse.e4.ui.workbench,  org.eclipse.e4.ui.workbench.renderers.swt,  org.eclipse.e4.ui.workbench.swt,  org.eclipse.ui.workbench,  org.eclipse.e4.core.commands"
> [ERROR]                 Unresolved requirement: Import-Package: org.eclipse.e4.ui.internal.workbench
> [ERROR]                   -> Export-Package: org.eclipse.e4.ui.internal.workbench; bundle-symbolic-name="org.eclipse.e4.ui.workbench"; bundle-version="1.13.0.v20210521-0741"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.workbench.fragment,  org.eclipse.e4.ui.workbench.renderers.swt,  org.eclipse.e4.ui.workbench.swt,  org.eclipse.ui.workbench,  org.eclipse.ui.ide.application,  org.eclipse.ui.ide,  org.eclipse.e4.emf.xpath.test"
> [ERROR]                 Unresolved requirement: Import-Package: org.eclipse.e4.core.commands.internal
> [ERROR]                   -> Export-Package: org.eclipse.e4.core.commands.internal; bundle-symbolic-name="org.eclipse.e4.core.commands"; bundle-version="1.0.0.v20210507-1901"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.bindings,org.eclipse.ui.workbench"
> [ERROR] -> [Help 1]
> org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: org.osgi.framework.BundleException: Bundle example-bundle cannot be resolved:example-bundle [68]
>   Unresolved requirement: Require-Bundle: org.eclipse.ui; bundle-version="3.6.0"
>     -> Bundle-SymbolicName: org.eclipse.ui; bundle-version="3.119.100.v20210830-0942"; singleton:="true"
>        org.eclipse.ui [5]
>          Unresolved requirement: Require-Bundle: org.eclipse.ui.workbench; bundle-version="[3.120.0,4.0.0)"; visibility:="reexport"
>            -> Bundle-SymbolicName: org.eclipse.ui.workbench; bundle-version="3.123.0.v20210817-0704"; singleton:="true"
>               org.eclipse.ui.workbench [64]
>                 Unresolved requirement: Import-Package: org.eclipse.e4.ui.internal.workbench.addons
>                   -> Export-Package: org.eclipse.e4.ui.internal.workbench.addons; bundle-symbolic-name="org.eclipse.e4.ui.workbench"; bundle-version="1.13.0.v20210521-0741"; version="0.0.0"; x-internal:="true"
>                      org.eclipse.e4.ui.workbench [50]
>                        Unresolved requirement: Require-Bundle: org.eclipse.e4.ui.model.workbench; bundle-version="1.2.0"
>                          -> Bundle-SymbolicName: org.eclipse.e4.ui.model.workbench; bundle-version="2.2.0.v20210727-1533"; singleton:="true"
>                             org.eclipse.e4.ui.model.workbench [48]
>                               Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
>                                 -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
>                                    org.eclipse.e4.core.services [32]
>                                      Unresolved requirement: Require-Capability: osgi.extender; filter:="(&(osgi.extender=osgi.component)(version>=1.2)(!(version>=2.0)))"
>                        Unresolved requirement: Require-Bundle: org.eclipse.e4.ui.services; bundle-version="0.9.0"
>                          -> Bundle-SymbolicName: org.eclipse.e4.ui.services; bundle-version="1.5.0.v20210115-1333"; singleton:="true"
>                             org.eclipse.e4.ui.services [41]
>                               Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
>                                 -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
>                        Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
>                          -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
>                 Unresolved requirement: Import-Package: org.eclipse.e4.core.commands
>                   -> Export-Package: org.eclipse.e4.core.commands; bundle-symbolic-name="org.eclipse.e4.core.commands"; bundle-version="1.0.0.v20210507-1901"; version="0.0.0"
>                      org.eclipse.e4.core.commands [28]
>                        Unresolved requirement: Import-Package: org.eclipse.e4.core.services.log
>                          -> Export-Package: org.eclipse.e4.core.services.log; bundle-symbolic-name="org.eclipse.e4.core.services"; bundle-version="2.3.0.v20210723-1034"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.bindings,  org.eclipse.e4.ui.services,  org.eclipse.e4.ui.workbench,  org.eclipse.e4.ui.workbench.renderers.swt,  org.eclipse.e4.ui.workbench.swt,  org.eclipse.ui.workbench,  org.eclipse.e4.core.commands"
>                 Unresolved requirement: Import-Package: org.eclipse.e4.ui.internal.workbench
>                   -> Export-Package: org.eclipse.e4.ui.internal.workbench; bundle-symbolic-name="org.eclipse.e4.ui.workbench"; bundle-version="1.13.0.v20210521-0741"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.workbench.fragment,  org.eclipse.e4.ui.workbench.renderers.swt,  org.eclipse.e4.ui.workbench.swt,  org.eclipse.ui.workbench,  org.eclipse.ui.ide.application,  org.eclipse.ui.ide,  org.eclipse.e4.emf.xpath.test"
>                 Unresolved requirement: Import-Package: org.eclipse.e4.core.commands.internal
>                   -> Export-Package: org.eclipse.e4.core.commands.internal; bundle-symbolic-name="org.eclipse.e4.core.commands"; bundle-version="1.0.0.v20210507-1901"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.bindings,org.eclipse.ui.workbench"
> 
>     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:120)
>     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
>     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
>     at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
>     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
>     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
>     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke (Method.java:566)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
> Caused by: java.lang.RuntimeException: org.osgi.framework.BundleException: Bundle example-bundle cannot be resolved:example-bundle [68]
>   Unresolved requirement: Require-Bundle: org.eclipse.ui; bundle-version="3.6.0"
>     -> Bundle-SymbolicName: org.eclipse.ui; bundle-version="3.119.100.v20210830-0942"; singleton:="true"
>        org.eclipse.ui [5]
>          Unresolved requirement: Require-Bundle: org.eclipse.ui.workbench; bundle-version="[3.120.0,4.0.0)"; visibility:="reexport"
>            -> Bundle-SymbolicName: org.eclipse.ui.workbench; bundle-version="3.123.0.v20210817-0704"; singleton:="true"
>               org.eclipse.ui.workbench [64]
>                 Unresolved requirement: Import-Package: org.eclipse.e4.ui.internal.workbench.addons
>                   -> Export-Package: org.eclipse.e4.ui.internal.workbench.addons; bundle-symbolic-name="org.eclipse.e4.ui.workbench"; bundle-version="1.13.0.v20210521-0741"; version="0.0.0"; x-internal:="true"
>                      org.eclipse.e4.ui.workbench [50]
>                        Unresolved requirement: Require-Bundle: org.eclipse.e4.ui.model.workbench; bundle-version="1.2.0"
>                          -> Bundle-SymbolicName: org.eclipse.e4.ui.model.workbench; bundle-version="2.2.0.v20210727-1533"; singleton:="true"
>                             org.eclipse.e4.ui.model.workbench [48]
>                               Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
>                                 -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
>                                    org.eclipse.e4.core.services [32]
>                                      Unresolved requirement: Require-Capability: osgi.extender; filter:="(&(osgi.extender=osgi.component)(version>=1.2)(!(version>=2.0)))"
>                        Unresolved requirement: Require-Bundle: org.eclipse.e4.ui.services; bundle-version="0.9.0"
>                          -> Bundle-SymbolicName: org.eclipse.e4.ui.services; bundle-version="1.5.0.v20210115-1333"; singleton:="true"
>                             org.eclipse.e4.ui.services [41]
>                               Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
>                                 -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
>                        Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
>                          -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
>                 Unresolved requirement: Import-Package: org.eclipse.e4.core.commands
>                   -> Export-Package: org.eclipse.e4.core.commands; bundle-symbolic-name="org.eclipse.e4.core.commands"; bundle-version="1.0.0.v20210507-1901"; version="0.0.0"
>                      org.eclipse.e4.core.commands [28]
>                        Unresolved requirement: Import-Package: org.eclipse.e4.core.services.log
>                          -> Export-Package: org.eclipse.e4.core.services.log; bundle-symbolic-name="org.eclipse.e4.core.services"; bundle-version="2.3.0.v20210723-1034"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.bindings,  org.eclipse.e4.ui.services,  org.eclipse.e4.ui.workbench,  org.eclipse.e4.ui.workbench.renderers.swt,  org.eclipse.e4.ui.workbench.swt,  org.eclipse.ui.workbench,  org.eclipse.e4.core.commands"
>                 Unresolved requirement: Import-Package: org.eclipse.e4.ui.internal.workbench
>                   -> Export-Package: org.eclipse.e4.ui.internal.workbench; bundle-symbolic-name="org.eclipse.e4.ui.workbench"; bundle-version="1.13.0.v20210521-0741"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.workbench.fragment,  org.eclipse.e4.ui.workbench.renderers.swt,  org.eclipse.e4.ui.workbench.swt,  org.eclipse.ui.workbench,  org.eclipse.ui.ide.application,  org.eclipse.ui.ide,  org.eclipse.e4.emf.xpath.test"
>                 Unresolved requirement: Import-Package: org.eclipse.e4.core.commands.internal
>                   -> Export-Package: org.eclipse.e4.core.commands.internal; bundle-symbolic-name="org.eclipse.e4.core.commands"; bundle-version="1.0.0.v20210507-1901"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.bindings,org.eclipse.ui.workbench"
> 
>     at org.eclipse.tycho.core.osgitools.OsgiBundleProject.getResolverState (OsgiBundleProject.java:329)
>     at org.eclipse.tycho.core.osgitools.OsgiBundleProject.resolveClassPath (OsgiBundleProject.java:196)
>     at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject (DefaultTychoResolver.java:173)
>     at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.lambda$resolveProjects$0 (TychoMavenLifecycleParticipant.java:147)
>     at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:183)
>     at java.util.stream.WhileOps$1$1.accept (WhileOps.java:99)
>     at java.util.ArrayList$ArrayListSpliterator.tryAdvance (ArrayList.java:1632)
>     at java.util.stream.ReferencePipeline.forEachWithCancel (ReferencePipeline.java:127)
>     at java.util.stream.AbstractPipeline.copyIntoWithCancel (AbstractPipeline.java:502)
>     at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:488)
>     at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:474)
>     at java.util.stream.ForEachOps$ForEachOp.evaluateSequential (ForEachOps.java:150)
>     at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential (ForEachOps.java:173)
>     at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
>     at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:497)
>     at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.resolveProjects (TychoMavenLifecycleParticipant.java:178)
>     at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead (TychoMavenLifecycleParticipant.java:105)
>     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
>     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
>     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
>     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
>     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
>     at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
>     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
>     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
>     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke (Method.java:566)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
> Caused by: org.osgi.framework.BundleException: Bundle example-bundle cannot be resolved:example-bundle [68]
>   Unresolved requirement: Require-Bundle: org.eclipse.ui; bundle-version="3.6.0"
>     -> Bundle-SymbolicName: org.eclipse.ui; bundle-version="3.119.100.v20210830-0942"; singleton:="true"
>        org.eclipse.ui [5]
>          Unresolved requirement: Require-Bundle: org.eclipse.ui.workbench; bundle-version="[3.120.0,4.0.0)"; visibility:="reexport"
>            -> Bundle-SymbolicName: org.eclipse.ui.workbench; bundle-version="3.123.0.v20210817-0704"; singleton:="true"
>               org.eclipse.ui.workbench [64]
>                 Unresolved requirement: Import-Package: org.eclipse.e4.ui.internal.workbench.addons
>                   -> Export-Package: org.eclipse.e4.ui.internal.workbench.addons; bundle-symbolic-name="org.eclipse.e4.ui.workbench"; bundle-version="1.13.0.v20210521-0741"; version="0.0.0"; x-internal:="true"
>                      org.eclipse.e4.ui.workbench [50]
>                        Unresolved requirement: Require-Bundle: org.eclipse.e4.ui.model.workbench; bundle-version="1.2.0"
>                          -> Bundle-SymbolicName: org.eclipse.e4.ui.model.workbench; bundle-version="2.2.0.v20210727-1533"; singleton:="true"
>                             org.eclipse.e4.ui.model.workbench [48]
>                               Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
>                                 -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
>                                    org.eclipse.e4.core.services [32]
>                                      Unresolved requirement: Require-Capability: osgi.extender; filter:="(&(osgi.extender=osgi.component)(version>=1.2)(!(version>=2.0)))"
>                        Unresolved requirement: Require-Bundle: org.eclipse.e4.ui.services; bundle-version="0.9.0"
>                          -> Bundle-SymbolicName: org.eclipse.e4.ui.services; bundle-version="1.5.0.v20210115-1333"; singleton:="true"
>                             org.eclipse.e4.ui.services [41]
>                               Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
>                                 -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
>                        Unresolved requirement: Require-Bundle: org.eclipse.e4.core.services; bundle-version="0.9.0"
>                          -> Bundle-SymbolicName: org.eclipse.e4.core.services; bundle-version="2.3.0.v20210723-1034"; singleton:="true"
>                 Unresolved requirement: Import-Package: org.eclipse.e4.core.commands
>                   -> Export-Package: org.eclipse.e4.core.commands; bundle-symbolic-name="org.eclipse.e4.core.commands"; bundle-version="1.0.0.v20210507-1901"; version="0.0.0"
>                      org.eclipse.e4.core.commands [28]
>                        Unresolved requirement: Import-Package: org.eclipse.e4.core.services.log
>                          -> Export-Package: org.eclipse.e4.core.services.log; bundle-symbolic-name="org.eclipse.e4.core.services"; bundle-version="2.3.0.v20210723-1034"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.bindings,  org.eclipse.e4.ui.services,  org.eclipse.e4.ui.workbench,  org.eclipse.e4.ui.workbench.renderers.swt,  org.eclipse.e4.ui.workbench.swt,  org.eclipse.ui.workbench,  org.eclipse.e4.core.commands"
>                 Unresolved requirement: Import-Package: org.eclipse.e4.ui.internal.workbench
>                   -> Export-Package: org.eclipse.e4.ui.internal.workbench; bundle-symbolic-name="org.eclipse.e4.ui.workbench"; bundle-version="1.13.0.v20210521-0741"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.workbench.fragment,  org.eclipse.e4.ui.workbench.renderers.swt,  org.eclipse.e4.ui.workbench.swt,  org.eclipse.ui.workbench,  org.eclipse.ui.ide.application,  org.eclipse.ui.ide,  org.eclipse.e4.emf.xpath.test"
>                 Unresolved requirement: Import-Package: org.eclipse.e4.core.commands.internal
>                   -> Export-Package: org.eclipse.e4.core.commands.internal; bundle-symbolic-name="org.eclipse.e4.core.commands"; bundle-version="1.0.0.v20210507-1901"; version="0.0.0"; x-friends:="org.eclipse.e4.ui.bindings,org.eclipse.ui.workbench"
> 
>     at org.eclipse.tycho.core.osgitools.EquinoxResolver.assertResolved (EquinoxResolver.java:510)
>     at org.eclipse.tycho.core.osgitools.EquinoxResolver.newResolvedState (EquinoxResolver.java:169)
>     at org.eclipse.tycho.core.osgitools.OsgiBundleProject.getResolverState (OsgiBundleProject.java:326)
>     at org.eclipse.tycho.core.osgitools.OsgiBundleProject.resolveClassPath (OsgiBundleProject.java:196)
>     at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject (DefaultTychoResolver.java:173)
>     at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.lambda$resolveProjects$0 (TychoMavenLifecycleParticipant.java:147)
>     at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:183)
>     at java.util.stream.WhileOps$1$1.accept (WhileOps.java:99)
>     at java.util.ArrayList$ArrayListSpliterator.tryAdvance (ArrayList.java:1632)
>     at java.util.stream.ReferencePipeline.forEachWithCancel (ReferencePipeline.java:127)
>     at java.util.stream.AbstractPipeline.copyIntoWithCancel (AbstractPipeline.java:502)
>     at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:488)
>     at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:474)
>     at java.util.stream.ForEachOps$ForEachOp.evaluateSequential (ForEachOps.java:150)
>     at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential (ForEachOps.java:173)
>     at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
>     at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:497)
>     at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.resolveProjects (TychoMavenLifecycleParticipant.java:178)
>     at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead (TychoMavenLifecycleParticipant.java:105)
>     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
>     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
>     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
>     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
>     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
>     at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
>     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
>     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
>     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke (Method.java:566)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
> [ERROR] 
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR] 
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/InternalErrorException
> 
</details>

Any ideas?
Is this a nexus or tycho related issue?

Regards,
Ahmet

**Comment by Bananeweizen** [comment]
Can you try Tycho 2.3.0 and the original Eclipse P2 URL? If that one works, then it's surely the Nexus mirror. That being said, I had never issues with using Nexus as internal mirror. However, I can't test that anymore, since we switched to Artifactory meanwhile.
